src/providers/CPKEthersProvider.ts

```diff
@@ -85,28 +101,28 @@ class CPKEthersProvider extends CPKProvider {
     };
   }
 
-  getProvider() {
+  getProvider(): any {
     // eslint-disable-next-line no-underscore-dangle
     return this.signer.provider.provider || this.signer.provider._web3Provider;
   }
 
-  async getNetworkId() {
+  async getNetworkId(): Promise<number> {
     return (await this.signer.provider.getNetwork()).chainId;
   }
 
-  async getOwnerAccount() {
+  async getOwnerAccount(): Promise<string> {
     return this.signer.getAddress();
   }
 
-  async getCodeAtAddress(contract) {
-    return this.signer.provider.getCode(this.constructor.getContractAddress(contract));
+  async getCodeAtAddress(contract: any): Promise<string> {
+    return this.signer.provider.getCode(this.getContractAddress(contract));
   }
 
-  static getContractAddress(contract) {
+  getContractAddress(contract: any): string {
     return contract.address;
   }
 
-  checkSingleCall(from, to, value, data) {
+  checkSingleCall(from: string, to: string, value: number, data: string): Promise<any> {
```

Comment:
not sure if `number` is the best type for `value` as this might trigger some issue on higher values. Maybe we should allow number or string.